lambdas/classifier/app.py

```diff
@@ -0,0 +1,99 @@
+import itertools
+import json
+import re
+from datetime import datetime
+
+import boto3
+
+MANIFEST_PREFIX_V1 = ".quilt/packages/"
+POINTER_PREFIX_V1 = ".quilt/named_packages/"
+
+
+event_bridge = boto3.client('events')
+s3 = boto3.client('s3')
+
+
+class EventsQueue:
+    MAX_SIZE = 10
+
+    def __init__(self):
+        self._events = []
+
+    def append(self, event):
+        self._events.append(event)
+        if len(self) >= self.MAX_SIZE:
+            self._flush()
+
+    def _flush(self):
+        events = self._events
+        self._events = []
+        event_bridge.put_events(Entries=events)
+        print(events)
+
+    def flush(self):
+        if self:
+            self._flush()
+
+    def __len__(self):
+        return len(self._events)
+
+    def __bool__(self):
+        return bool(self._events)
+
+
+PKG_POINTER_REGEX = re.compile(r"\.quilt/named_packages/([\w-]+/[\w-]+)/(.{10})")
+
+
+def pkg_created_event(s3_event):
+    print(s3_event)
+    if not s3_event['eventName'].startswith('ObjectCreated:'):
+        return
+    s3_event_obj = s3_event['s3']
+    obj = s3_event_obj['object']
+    key = obj['key']
+    match = PKG_POINTER_REGEX.fullmatch(key)
+    if not match:
+        return
+    pkg_name, pointer_name = match.groups()
+    if not '1451631600' <= pointer_name <= '1767250800':
+        return
+    bucket_obj = s3_event_obj['bucket']
+    bucket = bucket_obj['name']
+    etag = obj['eTag']
+    version_id = obj['versionId']
+    pkg_hash = s3.get_object(Bucket=bucket, Key=key, IfMatch=etag, VersionId=version_id)['Body'].read().decode()
+    manifest_key = MANIFEST_PREFIX_V1 + pkg_hash
+    resp = s3.head_object(Bucket=bucket, Key=manifest_key)
+    return {
```

Comment:
Very important: `eventVersion`. I suggest reviewing all S3 fields and copying where appropriate. https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html

Reply:
Also, we need an id field? Why not make it a hash of certain fields? We may need to dedupe messages based on this field.